app/schema/item.py

```diff
@@ -17,3 +15,48 @@ def construct_state(self):
 
     def get_state_class(self):
         pass
+
+    def validate(self, state):
+        if isinstance(state, self.get_state_class()):
+            is_valid = True
+            for child_state in state.children:
+                child_schema = self.questionnaire.get_item_by_id(child_state.id)
+                child_valid = child_schema.validate(child_state)
+                if not child_valid:
+                    is_valid = False
+
+            return is_valid
+        else:
+            import pdb
+            pdb.set_trace()
+            raise Exception('Cannot validate - incorrect state class')
+
+    def augment_with_state(self, state):
+        if state.id == self.id:
+            self.is_valid = state.is_valid
+            self.errors = state.errors
+            self.warnings = state.warnings
+            for child_state in state.children:
+                child_schema = self.questionnaire.get_item_by_id(child_state.id)
+                if child_schema:
+                    child_schema.augment_with_state(child_state)
+                else:
```

Comment:
I think the state object should contain everything that is needed to render the page, and it would be better to expand that rather than the schema object.

Though after our discussion, that might be best left to the display widget branch. So maybe add a TODO here to remove this.